source/app/libpd.c

```diff
@@ -35,62 +35,65 @@ static void connect_parodus()
         WalInfo("max_retry_sleep is %d\n", max_retry_sleep );
 
 	    get_parodus_url(&parodus_url, &client_url);
-
-	    libpd_cfg_t cfg1 = {.service_name = "config",
-					.receive = true, .keepalive_timeout_secs = 64,
-					.parodus_url = parodus_url,
-					.client_url = client_url
-				   };
-                
-        WalPrint("libparodus_init with parodus url %s and client url %s\n",cfg1.parodus_url,cfg1.client_url);
-
-        while(1)
-        {
-                if(backoffRetryTime < max_retry_sleep)
-                {
-                      backoffRetryTime = (int) pow(2, c) -1;
-                }
-
-                WalPrint("New backoffRetryTime value calculated as %d seconds\n", backoffRetryTime);
-                int ret =libparodus_init (&current_instance, &cfg1);
-                WalPrint("ret is %d\n",ret);
-                if(ret ==0)
-                {
-                        WalInfo("Init for parodus Success..!!\n");
-                        WalInfo("WebPA is now ready to process requests\n");
-#ifdef RDKB_BUILD
-                        if (access("/tmp/webpa_start", F_OK) == -1 && errno == ENOENT)
-                        {
-                                system("print_uptime \"boot_to_WEBPA_READY_uptime\" \"/rdklogs/logs/WEBPAlog.txt.0\"");
-
-                                if((fd = creat("/tmp/webpa_start", S_IRUSR | S_IWUSR)) == -1)
-                                {
-                                        fprintf(stderr, "File: /tmp/webpa_start creation failed with error:%d\n", errno);
-                                }
-                                else
-                                {
-                                        close(fd);
-                                }
-                        }
-#endif
-                        break;
-                }
-                else
-                {
-                        WalError("Init for parodus failed: '%s'\n",libparodus_strerror(ret));
-                        sleep(backoffRetryTime);
-                        c++;
-
-                        if(backoffRetryTime == max_retry_sleep)
-                        {
-                        	c = 2;
-                        	backoffRetryTime = 0;
-                        	WalPrint("backoffRetryTime reached max value, reseting to initial value\n");
-                        }
-                }
-                retval = libparodus_shutdown(&current_instance);
-                WalPrint("libparodus_shutdown retval %d\n", retval);
-        }
+		
+		if(parodus_url != NULL && client_url != NULL)
+		{	
```

Comment:
I guess we need to free parodus_url and client_url once we break out of while loop

Reply:
After libparodus_init() we cannot free parodus_url and client_url as client reconnection is handled internally in libparodus which needs this urls later.